controllers/datadogagent/clusteragent.go

```diff
@@ -520,6 +520,27 @@ func newClusterAgentPodTemplate(logger logr.Logger, dda *datadoghqv1alpha1.Datad
 	return newPodTemplate, nil
 }
 
+func getPodAntiAffinity(affinity *corev1.Affinity) *corev1.Affinity {
```

Comment:
This function generates a default value which is [specific to the cluster agent](https://github.com/DataDog/datadog-operator/pull/381/files#diff-1ab2275c9afe43600275bda29e5e505bb4ecda385774675935de82f60e27a1b7R534),
```go
MatchLabels: map[string]string{
	datadoghqv1alpha1.AgentDeploymentComponentLabelKey: datadoghqv1alpha1.DefaultClusterAgentResourceSuffix,
},
```

It could be relevant to highlight its lack of genericity in its name.

```suggestion
func getClusterAgentAffinity(affinity *corev1.Affinity) *corev1.Affinity {
```

For consistency, it would be good to also rename the existing `getPodAffinity` function in [controllers/datadogagent/clusterchecksrunner.go](https://github.com/DataDog/datadog-operator/blob/2d350805e7a77421333789c48519649b3cf72d5d/controllers/datadogagent/clusterchecksrunner.go#L513) to highlight the fact it’s specific to cluster checks runners:

```diff
- // getPodAffinity returns the pod anti affinity of the cluster check runner pods
+ // getClusterChecksRunnerAffinity returns the pod anti affinity of the cluster check runner pods
  // the default anti affinity prefers scheduling the runners on different nodes if possible
  // for better checks stability in case of node failure.
- func getPodAffinity(affinity *corev1.Affinity) *corev1.Affinity {
+ func getClusterChecksRunnerAffinity(affinity *corev1.Affinity) *corev1.Affinity { 
```